Core/Utilities/TimeLeft/MJFTimeLeft.py

```diff
@@ -0,0 +1,93 @@
+########################################################################
+# $Id$
+########################################################################
+
+""" The Machine/Job Features TimeLeft utility interrogates the MJF values
+    for the current CPU and Wallclock consumed, as well as their limits.
+"""
+
+from DIRAC import gLogger, S_OK, S_ERROR
+
+__RCSID__ = "$Id$"
+
+import os, re, time, urllib
+
+class MJFTimeLeft:
+
+  #############################################################################
+  def __init__( self ):
+    """ Standard constructor
+    """
+    self.log = gLogger.getSubLogger( 'MJFTimeLeft' )
+    self.jobID = None
+    if os.environ.has_key( 'JOB_ID' ):
+      self.jobID = os.environ['JOB_ID']
+    self.queue = None
+    if os.environ.has_key( 'QUEUE' ):
+      self.queue = os.environ['QUEUE']
+
+    self.cpuLimit = None
+    self.wallClockLimit = None
+    self.log.verbose( 'jobID=%s, queue=%s' % ( self.jobID, self.queue ) )
```

Comment:
They're visible in the job parameters from the command line and in the web portal so they're very useful for debugging IMO.